test/framework/deployment_helpers.go

```diff
@@ -247,3 +282,128 @@ func WaitForDNSUpgrade(ctx context.Context, input WaitForDNSUpgradeInput, interv
 		return false, nil
 	}, intervals...).Should(BeTrue())
 }
+
+type AddUnevictablePodInput struct {
+	ClusterProxy                       ClusterProxy
```

Comment:
We can pass the workload cluster proxy instead of passing the management cluster proxy, because AddUnevictablePod only works on workload cluster.

So from the node_drain_timeout.go, workload cluster can be reached as follows:

```
		// Get a ClusterBroker so we can interact with the workload cluster
		selfHostedClusterProxy = input.BootstrapClusterProxy.GetWorkloadCluster(context.TODO(), cluster.Namespace, cluster.Name)

```

After this passing workload cluster client as ClusterProxy to this function as input should work.

Reply:
Thanks for your suggestion. I will make an update on it